runhouse/main.py

```diff
@@ -576,6 +592,95 @@ def status(
     _print_status(cluster_status, current_cluster)
 
 
+async def aget_clusters_from_den():
+    httpx_client = httpx.AsyncClient()
+
+    get_clusters_params = {"resource_type": "cluster", "folder": rns_client.username}
+    clusters_in_den_resp = await httpx_client.get(
+        f"{rns_client.api_server_url}/resource",
+        params=get_clusters_params,
+        headers=rns_client.request_headers(),
+    )
+
+    return clusters_in_den_resp
+
+
+def get_clusters_from_den():
+    return asyncio.run(aget_clusters_from_den())
+
+
+@cluster_app.command("list")
+def cluster_list():
+    """Load Runhouse clusters"""
+    import sky
+
+    # logged out case
+    if not rh.configs.token:
+        # TODO [SB]: adjust msg formatting (coloring etc)
+        sky_cli_command_formatted = f"{italic_bold_ansi}sky status -r{reset_format}"  # will be printed bold and italic
+        console.print(
+            f"Runhouse token required to view all clusters. Please run `runhouse login` to load your token. To view on-demand clusters run {sky_cli_command_formatted}"
+        )
+        return
+
+    sky_clusters = sky.status(refresh=True)
```

Comment:
any value to making `refresh` configurable here? if its True will query the cloud provider directly each time

Reply:
I think that if we won't call pass `refresh=True`, we might get results that are cached. That means that some clusters maybe shown as running, even thought they are not. Having said that, if there are running sky clusters which are not saved in den, we are displaying the following message to the users: `There are N live clusters that are not saved in Den. For more information, please run sky status -r`. So maybe it's fine not to pass refresh=True here. wdyt?

Reply:
yea i think as long as we're instructing the user to refresh with -r then its fine to not do so here